src/core/src/local_loggers/CompositeLogger.py

```diff
@@ -48,32 +51,74 @@ def log(self, message, message_type=Constants.TelemetryEventLevel.Informational)
             timestamp = self.env_layer.datetime.timestamp()
             self.file_logger.write("\n" + timestamp + "> " + message.strip(), fail_silently=False)
 
-    def log_error(self, message):
-        """log errors"""
+    def flush(self, message, message_level):
+        if message_level == Constants.MessageLevel.ERROR:
+            self.flush_error(message)
+        elif message_level == Constants.MessageLevel.WARNING:
+            self.flush_warning(message)
+        elif message_level == Constants.MessageLevel.DEBUG:
+            self.flush_debug(message)
+        elif message_level == Constants.MessageLevel.VERBOSE:
+            self.flush_verbose(message)
+        self.buffer_message = ""
+
+    def buffer_or_flush_message(self, message, buffer, message_level):
         message = self.__remove_substring_from_message(message, Constants.ERROR_ADDED_TO_STATUS)
+
+        if buffer == Constants.BufferMessage.FALSE:
+            if len(self.buffer_message) > 0:
```

Comment:
You could simple append to buffer and do one flush here. Not need to check size of the buffer

Reply:
Adding to previous point, this will make this option suitable for FLUSH enum type too. So for both options FLUSH and FALSE, you will append to buffer and write. For option TRUE, you can just store in buffer.